crates/bevy_text/src/text2d_picking_backend.rs

```diff
@@ -0,0 +1,176 @@
+//! A [`bevy_picking`] backend for [`Text2d`]. Works for 2d text with arbitrary transforms.
+//! Picking is done based on [`TextLayoutInfo`] bounds, not visible pixels. This means that
+//! 2d text is pickable even in its transparent areas.
+//!
+//! **Note:** This backend is pretty much a 1:1 port of the [`bevy_sprite`] picking backend,
+//! and *is not responsible for handling the picking of UI Text*. For that, please refer to the
+//! picking backend implementation under the `bevy_ui` crate.
+
+use core::cmp::Reverse;
+
+use crate::{Text2d, TextLayoutInfo};
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_math::{prelude::*, FloatExt, FloatOrd};
+use bevy_picking::backend::prelude::*;
+use bevy_render::prelude::*;
+use bevy_sprite::Anchor;
+use bevy_transform::prelude::*;
+use bevy_window::PrimaryWindow;
+
+#[derive(Clone)]
+pub struct Text2dPickingPlugin;
+
+impl Plugin for Text2dPickingPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(PreUpdate, text2d_picking.in_set(PickSet::Backend));
+    }
+}
+
+pub fn text2d_picking(
+    pointers: Query<(&PointerId, &PointerLocation)>,
+    cameras: Query<(Entity, &Camera, &GlobalTransform, &OrthographicProjection)>,
+    primary_window: Query<Entity, With<PrimaryWindow>>,
+    text2d_query: Query<
+        (
+            Entity,
+            &TextLayoutInfo,
+            &Anchor,
+            &GlobalTransform,
+            Option<&PickingBehavior>,
+            &ViewVisibility,
+        ),
+        With<Text2d>,
+    >,
+    mut output: EventWriter<PointerHits>,
+) {
+    let mut sorted_text2d: Vec<_> = text2d_query
+        .iter()
+        .filter_map(
+            |(entity, text_layout_info, anchor, transform, picking_behavior, vis)| {
+                if !transform.affine().is_nan() && vis.get() {
+                    Some((
+                        entity,
+                        text_layout_info,
+                        anchor,
+                        transform,
+                        picking_behavior,
+                    ))
+                } else {
+                    None
+                }
+            },
+        )
+        .collect();
+
+    sorted_text2d.sort_by_key(|x| Reverse(FloatOrd(x.3.translation().z)));
```

Comment:
Can you update this to match #17174 ? I would be shocked if the perf characteristics weren't the same.